src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -1184,11 +1184,24 @@ private void ExecuteAndCatchIfThrows_NoLock<TArg>(DiagnosticAnalyzer analyzer, A
                 cancellationToken.ThrowIfCancellationRequested();
                 analyze(argument);
             }
-            catch (Exception ex) when (HandleAnalyzerException(ex, analyzer, info, OnAnalyzerException, _analyzerExceptionFilter, cancellationToken))
+            catch (Exception ex) when (HandleAnalyzerException(analyzer, ex, info) &&
+                HandleAnalyzerException(ex, analyzer, info, OnAnalyzerException, _analyzerExceptionFilter, cancellationToken))
             {
             }
         }
 
+        private bool HandleAnalyzerException(DiagnosticAnalyzer analyzer, Exception ex, in AnalysisContextInfo? info)
+        {
+            if (this.Compilation.FeatureDebugAnalyzers)
+            {
+                Debug.Assert(false);
+                Environment.FailFast(CreateAnalyzerExceptionDiagnostic(analyzer, ex, info).ToString());
```

Comment:
>Environment.FailFast(CreateAnalyzerExceptionDiagnostic(analyzer, ex, info).ToString()); [](http://example.com/codeflow?start=16&length=87)

Is this really going to produce a Watson dump with the stack at the position that threw the exception and with the right exception information for it?

Reply:
As long as this is called in the `when` then it will produce a Watson with failure on the stack.